### ISSUE TITLE
Bundler using wrong ruby version

### ISSUE THREAD

**Issue by hackerkid** [bundler/bundler#4260]
There are million questions on SOF regarding this. But I was not able to figure out a solution. 

 I have ruby 2.3.0 installed in my system. The output of ruby -v is

`    ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-linux]`

Output of rbenv global

```
2.3.0
```
Output of rbenv local

```
2.3.0
```

           

But when I try to run rails using the below command.

> bundle exec rails new ~/my-test-app --dev

I am getting the following error

    Your Ruby version is 2.1.7, but your Gemfile specified 2.3.0

I have both Ruby versions installed. How to make bundle use ruby 2.3.0 instead of 2.1.7

**Comment by segiddins** [comment]
Do you happen to have an executable named `rails` on your `PATH` by any chance? What's the output of `bundle env`?

**Comment by hackerkid** [comment]
The output is 
```
Environment

    Bundler   1.11.2
    Rubygems  2.5.1
    Ruby      2.1.7p400 (2015-08-18 revision 51632) [x86_64-linux-gnu]
    GEM_HOME  /home/vishnu/.rvm/gems/ruby-2.3.0
    GEM_PATH  /home/vishnu/.rvm/gems/ruby-2.3.0:/home/vishnu/.rvm/gems/ruby-2.3.0@global
    RVM       1.26.11 (latest)
    Git       1.9.1

```

Why is the version showing `2.1.7` ?

**Comment by hackerkid** [comment]
The contents of $PATH 

```
/home/vishnu/.rbenv/plugins/ruby-build/bin:/home/vishnu/.rbenv/shims:/home/vishnu/.rbenv/bin:/usr/local/heroku/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/vishnu/.rvm/bin
```

**Comment by TimMoore** [comment]
@hackerkid this usually happens when you're running a version of Bundler that was installed using a different Ruby version than the first one on your path, and it has a path to that version of Ruby hardcoded in the `#!` line of the script file. What does this print? `head -n1 $(which bundle)`

You can usually fix this by reinstalling Bundler (run `gem install bundler`) but I would recommend simplifying things to avoid this kind of situation. It looks like you have both rbenv and rvm directories on your path. Are you using both of these? I'd suggest picking one and uninstalling the other.

**Comment by hackerkid** [comment]
@TimMoore It was priting 2.1 instead of 2.3. I was not able to fix the error by re-installing bundler. The error was fixed by updating the `$(which bundle)` file ruby version from 2.1 to 2.3. 

Shouldn't be there an option to change the ruby version bundler use?

**Comment by segiddins** [comment]
Why are your binstubs hardcoding in a ruby version instead of picking the ruby on your `path`?

**Comment by hackerkid** [comment]
@segiddins I am not sure. I will update if I figure it out. Closing the issue now.

**Comment by CarefreeSlacker** [comment]
Also you can integrate bundler with RVM.
https://rvm.io/integration/bundler
It will allow you to switch between different ruby versions easily.

**Comment by bretonio** [comment]
I uninstalled 'rvm' and installed 2.3.0 with `rbenv` with the same results as the fellow who started this thread. 

But `bundle env` returned 
```
usr/local/bin/bundle: /usr/bin/ruby1.9.1: bad interpreter: No such file or directory
```

Ruby 1.9.1 is the version installed with `apt-get install ruby` -- this is a Linux Mint machine. 

The thing is that I did an `apt-get remove ruby` already. 

'rbenv' was installed from source.

Excuse me for thinking aloud here. 

I have resolved the issue apparently with `rbenv global 2.3.0` followed by `rbenv rehash` ... Thanks!

**Comment by plinaky** [comment]
Dear @hackerkid @TimMoore

I am stuck there

$ rbenv rehash
$ gem uninstall bundler
$ ruby -v
ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-linux]
$ rbenv global
2.4.0
$ rbenv global 2.4.0
$ rbenv rehash
$ gem install bundler
Fetching: bundler-1.13.7.gem (100%)
Successfully installed bundler-1.13.7
Parsing documentation for bundler-1.13.7
Installing ri documentation for bundler-1.13.7
Done installing documentation for bundler after 4 seconds
1 gem installed
$ ruby -v
ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-linux]
$ bundle env
Environment

    Bundler   1.13.7
    Rubygems  1.8.23
    Ruby      1.9.3p484 (2013-11-22 revision 43786) [x86_64-linux]
    Git       1.9.1

Gemfile

    source 'https://rubygems.org'
    gem 'github-pages', group: :jekyll_plugins

Gemfile.lock

    <No /XXX/Gemfile.lock found>

------------------------------------------------------------------
I do not understand hwo you do
"The error was fixed by updating the $(which bundle) file ruby version from 2.1 to 2.3."

Apologies for asking again

------------------------------------------------------------------
PS
ruby 2.4.0 was installed in this way
ruby-install --install-dir ~/.rbenv/versions/2.4.0 ruby 2.4.0

and ruby-install was installed from shell

I am using zsh and the same error occured using bash

**Comment by TimMoore** [comment]
@plinaky that looks like a similar problem, but with the version of the `gem` command you're running. I think you're probably running the `gem` command installed for Ruby 1.9.3 rather than the one in your `rbenv` installation of 2.4.0.

Here are some things you can check:

- `which gem` should return the path to your rbenv shims directory (see https://github.com/rbenv/rbenv#understanding-shims)
- `rbenv which gem` — does it print a path inside `~/.rbenv/versions/2.4.0` or something like `/usr/bin/gem`?
- `which -a gem` prints all of the gem commands that can be found on the path
- `rbenv which bundle` prints the path to the bundle command that rbenv is using
- `echo $PATH` prints the search order used to search for commands: if you have set up `rbenv` correctly, it should be adding its shims directory to the front of your path, but if something else such as `/usr/bin` or `/usr/local/bin` is before it, then you'll get the wrong version of commands

**Comment by plinaky** [comment]
Thank you for feedback
Here is the output of commands

$ which gem
/usr/bin/gem
$ rbenv which gem
/home/cabv/.rbenv/versions/2.4.0/bin/gem
$ which -a gem
/usr/bin/gem
/usr/bin/gem
$ rbenv which bundle
/home/cabv/.rbenv/versions/2.4.0/bin/bundle
$ echo $PATH
/home/cabv/.rbenv/versions/2.4.0/bin:....

I had to update $PATH variable manually

**Comment by plinaky** [comment]
@TimMoore

Sorry. I know what I have done is dirty, just to keep trace for those who try the same.

sudo ln -s /home/cabv/.rbenv/versions/2.4.0/bin/bundle /usr/local/bin/bundle

Then I have new issues.

$ bundle install
...
Using bundler 1.13.7
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
/home/cabv/.rbenv/versions/2.4.0/bin/ruby -r ./siteconf20170102-23567-e40mc6.rb extconf.rb
creating Makefile

current directory: /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
make "DESTDIR=" clean

current directory: /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
make "DESTDIR="
compiling generator.c
generator.c: In function ‘generate_json’:
generator.c:861:25: error: ‘rb_cFixnum’ undeclared (first use in this function)
     } else if (klass == rb_cFixnum) {
                         ^
generator.c:861:25: note: each undeclared identifier is reported only once for each function it appears in
generator.c:863:25: error: ‘rb_cBignum’ undeclared (first use in this function)
     } else if (klass == rb_cBignum) {
                         ^
generator.c: At top level:
cc1: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-constant-logical-operand" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-parentheses-equality" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-tautological-compare" [enabled by default]
make: *** [generator.o] Error 1

make failed, exit code 2

Gem files will remain installed in /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/json-1.8.3 for inspection.
Results logged to /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0-static/json-1.8.3/gem_make.out

An error occurred while installing json (1.8.3), and Bundler cannot continue.
Make sure that `gem install json -v '1.8.3'` succeeds before bundling.



and




$ gem install json -v '1.8.3'                                                                                                                                                                    ──(\o_ 5 _o/)─┘
Building native extensions.  This could take a while...
ERROR:  Error installing json:
	ERROR: Failed to build gem native extension.

    current directory: /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
/home/cabv/.rbenv/versions/2.4.0/bin/ruby -r ./siteconf20170102-23908-8xb846.rb extconf.rb
creating Makefile

current directory: /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
make "DESTDIR=" clean

current directory: /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
make "DESTDIR="
compiling generator.c
generator.c: In function ‘generate_json’:
generator.c:861:25: error: ‘rb_cFixnum’ undeclared (first use in this function)
     } else if (klass == rb_cFixnum) {
                         ^
generator.c:861:25: note: each undeclared identifier is reported only once for each function it appears in
generator.c:863:25: error: ‘rb_cBignum’ undeclared (first use in this function)
     } else if (klass == rb_cBignum) {
                         ^
generator.c: At top level:
cc1: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-constant-logical-operand" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-parentheses-equality" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-tautological-compare" [enabled by default]
make: *** [generator.o] Error 1

make failed, exit code 2

Gem files will remain installed in /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/gems/json-1.8.3 for inspection.
Results logged to /home/cabv/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0-static/json-1.8.3/gem_make.out

**Comment by TimMoore** [comment]
@plinaky the problem you're having isn't an issue with Bundler, so this is not really the right place to get help. It looks like you have a broken `rbenv` installation, and the workarounds you're using may be making the situation worse. I'd suggest starting over with your `rbenv` installation, making sure to follow all of the instructions in the `rbenv` README.

Alternatively, you might want to try a different Ruby version manager (I personally prefer [`chruby`](https://github.com/postmodern/chruby)).

**Comment by plinaky** [comment]
@TimMoore
As my 2 previous comments were trash, I have deleted them.
I have removed rbenv through apt-get and installed it manually by cloning from github.
It looks much better.

**Comment by davidmccoy** [comment]
Just thought I'd add that an `rbenv rehash` solved this issue for me when downgrading from Ruby 2.4 to 2.3.

**Comment by MahsaCharandabi** [comment]
@TimMoore I have the same problem with bundle env.
I checked all those 5 u mentioned, all of them were ok and the way it should be. but again i have problem on bundle env it is referring to ruby 2.0.0 while it is 2.4.0 in everywhere else. and it does not solve with reinstalling. is anything I can try?

**Comment by radarseesradar** [comment]
I had the same problem.  I'm running on ubuntu linux.  I created at .bash_profile file in my home directory, and put the following line in it: export PATH="$HOME/.rbenv/bin:$HOME/.rbenv/shims:$PATH".  I then added a line at the end of my .bashrc file to source my .bash_profile.  This worked for me.  Apparently, bundler launches a sub-shell under the cover, and the exports from .bashrc are not picked up by the sub-shell, but the ones in .bash_profile are.

**Comment by WAFulbright** [comment]
I am not able to have ruby 2.4.2 recognized, and it is installed it is installed /usr/bin/ruby.
Following is output for "bundle env" and "Gemfile:
I am stumped

 **bundle env
## Environment**

```
Bundler   1.15.4
Rubygems  2.0.14.1
Ruby      2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
Git       2.13.5 (Apple Git-94)
Platform  universal-darwin-16
OpenSSL   OpenSSL 0.9.8zc 19 Mar 2015
```

## Bundler settings

```
build.nokogiri
  Set for the current user (/Users/williamfulbright/.bundle/config): "--use-system-libraries"
```

## Gemfile

### Gemfile

```ruby
source 'https://rubygems.org'

ruby '2.0.0'

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '4.2.4'
# Use sqlite3 as the database for Active Record
gem 'sqlite3'
# Use SCSS for stylesheets
gem 'sass-rails', '~> 5.0'
# Use Uglifier as compressor for JavaScript assets
gem 'uglifier', '>= 1.3.0'
# Use CoffeeScript for .coffee assets and views
gem 'coffee-rails', '~> 4.1.0'
# See https://github.com/rails/execjs#readme for more supported runtimes
# gem 'therubyracer', platforms: :ruby

# Use jquery as the JavaScript library
gem 'jquery-rails'
# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
gem 'turbolinks'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.0'
# bundle exec rake doc:rails generates the API under doc/api.
gem 'sdoc', '~> 0.4.0', group: :doc

# Use ActiveModel has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Unicorn as the app server
# gem 'unicorn'

# Use Capistrano for deployment
# gem 'capistrano-rails', group: :development

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug'
end

group :development do
  # Access an IRB console on exception pages or by using <%= console %> in views
  gem 'web-console', '~> 2.0'

  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
end
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actionmailer (4.2.4)
      actionpack (= 4.2.4)
      actionview (= 4.2.4)
      activejob (= 4.2.4)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 1.0, >= 1.0.5)
    actionpack (4.2.4)
      actionview (= 4.2.4)
      activesupport (= 4.2.4)
      rack (~> 1.6)
      rack-test (~> 0.6.2)
      rails-dom-testing (~> 1.0, >= 1.0.5)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (4.2.4)
      activesupport (= 4.2.4)
      builder (~> 3.1)
      erubis (~> 2.7.0)
      rails-dom-testing (~> 1.0, >= 1.0.5)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    activejob (4.2.4)
      activesupport (= 4.2.4)
      globalid (>= 0.3.0)
    activemodel (4.2.4)
      activesupport (= 4.2.4)
      builder (~> 3.1)
    activerecord (4.2.4)
      activemodel (= 4.2.4)
      activesupport (= 4.2.4)
      arel (~> 6.0)
    activesupport (4.2.4)
      i18n (~> 0.7)
      json (~> 1.7, >= 1.7.7)
      minitest (~> 5.1)
      thread_safe (~> 0.3, >= 0.3.4)
      tzinfo (~> 1.1)
    arel (6.0.4)
    binding_of_caller (0.7.2)
      debug_inspector (>= 0.0.1)
    builder (3.2.3)
    byebug (8.2.2)
    coffee-rails (4.1.1)
      coffee-script (>= 2.2.0)
      railties (>= 4.0.0, < 5.1.x)
    coffee-script (2.4.1)
      coffee-script-source
      execjs
    coffee-script-source (1.12.2)
    concurrent-ruby (1.0.5)
    debug_inspector (0.0.3)
    erubis (2.7.0)
    execjs (2.7.0)
    globalid (0.3.6)
      activesupport (>= 4.1.0)
    i18n (0.8.6)
    jbuilder (2.4.1)
      activesupport (>= 3.0.0, < 5.1)
      multi_json (~> 1.2)
    jquery-rails (4.1.1)
      rails-dom-testing (>= 1, < 3)
      railties (>= 4.2.0)
      thor (>= 0.14, < 2.0)
    json (1.8.6)
    loofah (2.0.3)
      nokogiri (>= 1.5.9)
    mail (2.6.4)
      mime-types (>= 1.16, < 4)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mini_portile2 (2.0.0)
    minitest (5.10.3)
    multi_json (1.12.2)
    nokogiri (1.6.7.2)
      mini_portile2 (~> 2.0.0.rc2)
    rack (1.6.8)
    rack-test (0.6.3)
      rack (>= 1.0)
    rails (4.2.4)
      actionmailer (= 4.2.4)
      actionpack (= 4.2.4)
      actionview (= 4.2.4)
      activejob (= 4.2.4)
      activemodel (= 4.2.4)
      activerecord (= 4.2.4)
      activesupport (= 4.2.4)
      bundler (>= 1.3.0, < 2.0)
      railties (= 4.2.4)
      sprockets-rails
    rails-deprecated_sanitizer (1.0.3)
      activesupport (>= 4.2.0.alpha)
    rails-dom-testing (1.0.7)
      activesupport (>= 4.2.0.beta, < 5.0)
      nokogiri (~> 1.6.0)
      rails-deprecated_sanitizer (>= 1.0.1)
    rails-html-sanitizer (1.0.3)
      loofah (~> 2.0)
    railties (4.2.4)
      actionpack (= 4.2.4)
      activesupport (= 4.2.4)
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rake (12.1.0)
    rdoc (4.3.0)
    sass (3.4.21)
    sass-rails (5.0.4)
      railties (>= 4.0.0, < 5.0)
      sass (~> 3.1)
      sprockets (>= 2.8, < 4.0)
      sprockets-rails (>= 2.0, < 4.0)
      tilt (>= 1.1, < 3)
    sdoc (0.4.1)
      json (~> 1.7, >= 1.7.7)
      rdoc (~> 4.0)
    spring (1.6.4)
    sprockets (3.5.2)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.0.4)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    sqlite3 (1.3.13)
    thor (0.20.0)
    thread_safe (0.3.6)
    tilt (2.0.8)
    turbolinks (2.5.3)
      coffee-rails
    tzinfo (1.2.3)
      thread_safe (~> 0.1)
    uglifier (3.0.0)
      execjs (>= 0.3.0, < 3)
    web-console (2.3.0)
      activemodel (>= 4.0)
      binding_of_caller (>= 0.7.2)
      railties (>= 4.0)
      sprockets-rails (>= 2.0, < 4.0)

PLATFORMS
  ruby

DEPENDENCIES
  byebug
  coffee-rails (~> 4.1.0)
  jbuilder (~> 2.0)
  jquery-rails
  rails (= 4.2.4)
  sass-rails (~> 5.0)
  sdoc (~> 0.4.0)
  spring
  sqlite3
  turbolinks
  uglifier (>= 1.3.0)
  web-console (~> 2.0)

BUNDLED WITH
   1.11.2

 bundle env
## Environment

```
Bundler   1.15.4
Rubygems  2.0.14.1
Ruby      2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
Git       2.13.5 (Apple Git-94)
Platform  universal-darwin-16
OpenSSL   OpenSSL 0.9.8zc 19 Mar 2015
```

## Bundler settings

```
build.nokogiri
  Set for the current user (/Users/williamfulbright/.bundle/config): "--use-system-libraries"
```

**## Gemfile

### Gemfile**

```ruby
source 'https://rubygems.org'

ruby '2.0.0'

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '4.2.4'
# Use sqlite3 as the database for Active Record
gem 'sqlite3'
# Use SCSS for stylesheets
gem 'sass-rails', '~> 5.0'
# Use Uglifier as compressor for JavaScript assets
gem 'uglifier', '>= 1.3.0'
# Use CoffeeScript for .coffee assets and views
gem 'coffee-rails', '~> 4.1.0'
# See https://github.com/rails/execjs#readme for more supported runtimes
# gem 'therubyracer', platforms: :ruby

# Use jquery as the JavaScript library
gem 'jquery-rails'
# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
gem 'turbolinks'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.0'
# bundle exec rake doc:rails generates the API under doc/api.
gem 'sdoc', '~> 0.4.0', group: :doc

# Use ActiveModel has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Unicorn as the app server
# gem 'unicorn'

# Use Capistrano for deployment
# gem 'capistrano-rails', group: :development

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug'
end

group :development do
  # Access an IRB console on exception pages or by using <%= console %> in views
  gem 'web-console', '~> 2.0'

  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
end
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actionmailer (4.2.4)
      actionpack (= 4.2.4)
      actionview (= 4.2.4)
      activejob (= 4.2.4)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 1.0, >= 1.0.5)
    actionpack (4.2.4)
      actionview (= 4.2.4)
      activesupport (= 4.2.4)
      rack (~> 1.6)
      rack-test (~> 0.6.2)
      rails-dom-testing (~> 1.0, >= 1.0.5)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (4.2.4)
      activesupport (= 4.2.4)
      builder (~> 3.1)
      erubis (~> 2.7.0)
      rails-dom-testing (~> 1.0, >= 1.0.5)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    activejob (4.2.4)
      activesupport (= 4.2.4)
      globalid (>= 0.3.0)
    activemodel (4.2.4)
      activesupport (= 4.2.4)
      builder (~> 3.1)
    activerecord (4.2.4)
      activemodel (= 4.2.4)
      activesupport (= 4.2.4)
      arel (~> 6.0)
    activesupport (4.2.4)
      i18n (~> 0.7)
      json (~> 1.7, >= 1.7.7)
      minitest (~> 5.1)
      thread_safe (~> 0.3, >= 0.3.4)
      tzinfo (~> 1.1)
    arel (6.0.4)
    binding_of_caller (0.7.2)
      debug_inspector (>= 0.0.1)
    builder (3.2.3)
    byebug (8.2.2)
    coffee-rails (4.1.1)
      coffee-script (>= 2.2.0)
      railties (>= 4.0.0, < 5.1.x)
    coffee-script (2.4.1)
      coffee-script-source
      execjs
    coffee-script-source (1.12.2)
    concurrent-ruby (1.0.5)
    debug_inspector (0.0.3)
    erubis (2.7.0)
    execjs (2.7.0)
    globalid (0.3.6)
      activesupport (>= 4.1.0)
    i18n (0.8.6)
    jbuilder (2.4.1)
      activesupport (>= 3.0.0, < 5.1)
      multi_json (~> 1.2)
    jquery-rails (4.1.1)
      rails-dom-testing (>= 1, < 3)
      railties (>= 4.2.0)
      thor (>= 0.14, < 2.0)
    json (1.8.6)
    loofah (2.0.3)
      nokogiri (>= 1.5.9)
    mail (2.6.4)
      mime-types (>= 1.16, < 4)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mini_portile2 (2.0.0)
    minitest (5.10.3)
    multi_json (1.12.2)
    nokogiri (1.6.7.2)
      mini_portile2 (~> 2.0.0.rc2)
    rack (1.6.8)
    rack-test (0.6.3)
      rack (>= 1.0)
    rails (4.2.4)
      actionmailer (= 4.2.4)
      actionpack (= 4.2.4)
      actionview (= 4.2.4)
      activejob (= 4.2.4)
      activemodel (= 4.2.4)
      activerecord (= 4.2.4)
      activesupport (= 4.2.4)
      bundler (>= 1.3.0, < 2.0)
      railties (= 4.2.4)
      sprockets-rails
    rails-deprecated_sanitizer (1.0.3)
      activesupport (>= 4.2.0.alpha)
    rails-dom-testing (1.0.7)
      activesupport (>= 4.2.0.beta, < 5.0)
      nokogiri (~> 1.6.0)
      rails-deprecated_sanitizer (>= 1.0.1)
    rails-html-sanitizer (1.0.3)
      loofah (~> 2.0)
    railties (4.2.4)
      actionpack (= 4.2.4)
      activesupport (= 4.2.4)
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rake (12.1.0)
    rdoc (4.3.0)
    sass (3.4.21)
    sass-rails (5.0.4)
      railties (>= 4.0.0, < 5.0)
      sass (~> 3.1)
      sprockets (>= 2.8, < 4.0)
      sprockets-rails (>= 2.0, < 4.0)
      tilt (>= 1.1, < 3)
    sdoc (0.4.1)
      json (~> 1.7, >= 1.7.7)
      rdoc (~> 4.0)
    spring (1.6.4)
    sprockets (3.5.2)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.0.4)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    sqlite3 (1.3.13)
    thor (0.20.0)
    thread_safe (0.3.6)
    tilt (2.0.8)
    turbolinks (2.5.3)
      coffee-rails
    tzinfo (1.2.3)
      thread_safe (~> 0.1)
    uglifier (3.0.0)
      execjs (>= 0.3.0, < 3)
    web-console (2.3.0)
      activemodel (>= 4.0)
      binding_of_caller (>= 0.7.2)
      railties (>= 4.0)
      sprockets-rails (>= 2.0, < 4.0)

PLATFORMS
  ruby

DEPENDENCIES
  byebug
  coffee-rails (~> 4.1.0)
  jbuilder (~> 2.0)
  jquery-rails
  rails (= 4.2.4)
  sass-rails (~> 5.0)
  sdoc (~> 0.4.0)
  spring
  sqlite3
  turbolinks
  uglifier (>= 1.3.0)
  web-console (~> 2.0)

BUNDLED WITH
   1.11.2

**Comment by colby-swandale** [comment]
@WAFulbright please open a new issue with the details of your problem.

**Comment by passatgt** [comment]
For me, running `sudo aptitude purge ruby` and `gem install bundler` solved the issue. I'm using rvm.

**Comment by pocheptsov** [comment]
In my case, I had an existing $BUNDLE_PATH without enough permissions to the bundler user to write in.

> Your user account isn't allowed to install to the system RubyGems
> bundle install --path vendor/bundle

**Comment by JalenOng** [comment]
I have the same problem but I didn't see a clear solution to the problem in this thread?
My global and local version is `2.2.8` When I run `ruby -v`, my version is also `2.2.8`. But when I run bundle install, I get this error `Your Ruby version is 2.5.0, but your Gemfile specified 2.2.8
`. I have done rehash and reinstall bundler but nothing solves the problem. Bundle env shows this:
````
Bundler       1.16.1
  Platforms   ruby, x86_64-darwin-17
Ruby          2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin17]
  Full Path   /Users/jalenong/.asdf/installs/ruby/2.5.0/bin/ruby
  Config Dir  /Users/jalenong/.asdf/installs/ruby/2.5.0/etc
RubyGems      2.7.4
  Gem Home    /Users/jalenong/.asdf/installs/ruby/2.5.0/lib/ruby/gems/2.5.0
  Gem Path    /Users/jalenong/.gem/ruby/2.5.0:/Users/jalenong/.asdf/installs/ruby/2.5.0/lib/ruby/gems/2.5.0
  User Path   /Users/jalenong/.gem/ruby/2.5.0
  Bin Dir     /Users/jalenong/.asdf/installs/ruby/2.5.0/bin
Tools         
  Git         2.16.1
  RVM         not installed
  rbenv       rbenv 1.1.1

  chruby      not installed
```
`

**Comment by shivam-tripathi** [comment]
@JalenOng What is the output of `gem list` while you're in app directory?

**Comment by katrotz** [comment]
@JalenOng Had the same problem because I installed a new ruby version.
`gem install bundler` solved it.

**Comment by kirylpl** [comment]
Have the same issue

```
ruby -v
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-linux]

bundle
Your Ruby version is 2.3.1, but your Gemfile specified 2.3.0

rbenv which bundle
/home/user/.rbenv/versions/2.3.0/bin/bundle

```

**Comment by kirylpl** [comment]
`rbenv rehash` with removing previous /.rbenv/shims/.rbenv-shim worked for me

**Comment by Bellusterra** [comment]
Thank you @TimMoore

**Comment by jamesojo** [comment]
I fixed it by running `bundle update --ruby` more info [here](https://help.heroku.com/Q5ZF0DUO/ruby-build-fails-with-your-ruby-version-is-x-but-your-gemfile-specified-y)

**Comment by pllopis** [comment]
I am still seeing the issue that is mentioned further up in this page.
```
$ which ruby
/opt/rh/rh-ruby25/root/usr/bin/ruby
```
```
$ gem install bundler
```
```
$ head -1 $(which bundle)
#!/usr/bin/ruby
```
Bundler still defaults to the incorrect ruby path.

**Comment by camiloquezada89** [comment]
if you use Rbenv , do this: 

rbenv install -l
rbenv install 2.3.0 
rbenv local 2.3.0 
gem update --system
gem install bundler 
bundle update --bundler

**Comment by hackrzofheart** [comment]
Nothing will make it work unless you reinstall the metasploit-framework. I have been facing the issue for 1 week. I have personally tried and its running now.

$ pkg upgrade && pkg install git && pkg install curl && pkg install wget && pkg install nmap && curl -LO raw.githubusercontent.com/Hax4us/Metasploit_termux/master/metasploit.sh && chmod 777 metasploit.sh && ./metasploit.sh

**Comment by rodrigouchoa** [comment]
For anyone who comes across this issue, what solved for me was simply correcting the order of the path variable. I had to put `/usr/local/opt/ruby/bin` before `/usr/local/bin`:

`export PATH=$JAVA_HOME/bin:/usr/local/opt/ruby/bin:/usr/local/bin:$PATH`

**Comment by shivan** [comment]
In my case (no rbenv or rvm) I could solve the problem (it was using 2.3.7 instead of installed 2.5.5) by editing:

`/usr/local/bin/bundler` and `/usr/local/bin/bundle`

As these had in the first line of the script the wrong ruby version.

`#!/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby`

instead of correctly:

`#!/usr/local/opt/ruby@2.5/bin/ruby`

Even uninstall and reinstall of bundler (gem uninstall bundler, gem install bundler) and even with sudo it did not fix that. After I manually changed it, bundler now uses ruby 2.5.5.

**Comment by daniellizik** [comment]
After reading this thread and many other stackoverflow posts I'm still having issues with this on osx **catalina 10.15**

### 1. Check my brew openssl dir is correct

```
$ brew --prefix openssl
/usr/local/opt/openssl@1.1
```

### 2. Clean install of rvm with `--with-openssl-dir`

```
$ rm -rf /Users/daniellizik/.rvm/
$ rvm reinstall 2.1.3 --with-openssl-dir=`brew --prefix openssl`
```

### 3. Check my rubies and openssl again

```
$ rvm list
*  ruby 2.1.3
```
```
$ ruby -ropenssl -e "puts :OK"
OK
```

### 4. check `bundle env`

```yml
Bundler             1.17.2
  Platforms         ruby, universal-darwin-19
Ruby                2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
  Full Path         /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
  Config Dir        /Library/Ruby/Site
RubyGems            3.0.3
  Gem Home          /Users/daniellizik/.rvm/gems/ruby-2.1.3
  Gem Path          /Users/daniellizik/.rvm/gems/ruby-2.1.3:/Users/daniellizik/.rvm/gems/ruby-2.1.3@global
  User Path         /Users/daniellizik/.gem/ruby/2.6.0
  Bin Dir           /Users/daniellizik/.rvm/gems/ruby-2.1.3/bin
Tools               
  Git               2.20.1
  RVM               1.29.3 (1.29.3)
  rbenv             not installed
  chruby            not installed
  rubygems-bundler  (1.4.5)
```

### 5. switch ruby ver and check again

```
$ rvm use 2.1.3
Using /Users/daniellizik/.rvm/gems/ruby-2.1.3

$ ruby -ropenssl -e "puts :OK"
OK

$ ruby -v
ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-darwin19.0]

$ bundle install
Your Ruby version is 2.6.3, but your Gemfile specified ~> 2.1.3

$ head -n1 $(which bundle)
#!/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
```

Still not sure why even installing `rvm` with `--with-openssl-dir` flag is still using osx system ruby installation... 😭
